### DISCUSSION TITLE
Best way to initialize / create tables if not exists

### DISCUSSION THREAD

**Discussion by mathias22osterhagen22** [prisma/prisma#14040] [Q&A] [answered]
I'm using a docker-compose file to deploy a node.js app including a prisma schema and a PostgreSQL container.

## Solution 1
I'm currently writing a script to "initialize" / "create" the tables inside my PostgreSQL database but I didn't found in the documentation the best way to do it.

I was thinking to try any query and if it return an error `table not exists` starting `npx prisma db push`. 

## Solution 2
Or should I first create a `npx prisma migrate dev` that I'm naming `init` and use the `npx prisma migration deploy` as usual ?

(I suppose the solution 2 is the best one ?)

**Comment by austincrim** [comment, marked as answer]
Hey @mathias22osterhagen22 👋,

I think solution 2 is the best path forward. Create an initial migration and run `prisma migrate deploy` to get your database tables set up.

**Reply by mathias22osterhagen22** [reply]
It's what I was thinking too. Thank you :)

**Reply by LucasZapico** [reply]
Hey folks, I'm having a similar issue and I was wondering where you're running `npx prisma migrate dev --name init` in the docker spin up process? I throw it in my dockerfile where I generate my prisma client but it runs the command against `localhost:5432` which makes sense because it's running it against the image not the container.... I'm going to hack on it but I would be interested on your preferred pattern.  

I think I found a solution [prisma migrate deploy with docker - notiz.dev](https://notiz.dev/blog/prisma-migrate-deploy-with-docker)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
